conf/machine/imx6slevk.conf

```diff
@@ -13,14 +13,17 @@ KERNEL_DEVICETREE = "imx6sl-evk.dtb imx6sl-evk-csi.dtb imx6sl-evk-ldo.dtb \
                      imx6sl-evk-uart.dtb imx6sl-evk-btwifi.dtb"
 KERNEL_DEVICETREE_use-mainline-bsp = "imx6sl-evk.dtb"
 
-UBOOT_CONFIG ??= "sd"
-UBOOT_CONFIG[sd] = "mx6slevk_config,sdcard"
-UBOOT_CONFIG[epdc] = "mx6slevk_epdc_config"
-UBOOT_CONFIG[spinor] = "mx6slevk_spinor_config"
-UBOOT_CONFIG[mfgtool] = "mx6slevk_config"
+UBOOT_CONFIG ??= "${@bb.utils.contains('MACHINE_FEATURES', 'optee', 'sd-optee', 'sd', d)}"
```

Comment:
There is no need to remove the original `sd` config. You could do:

```
UBOOT_CONFIG ??= " \
    sd \
    ${@bb.utils.contains('MACHINE_FEATURES', 'optee', 'sd-optee', '', d)} \
"
```

However, we need to check if it works for both U-Boot versions: `u-boot-fslc` and `u-boot-imx`

If we don't support both, we need to do it conditionally otherwise we break the other build.

Reply:
Not supported by `u-boot-fslc`.

```
| *** Can't find default configuration "arch/../configs/mx6slevk_optee_defconfig"!
```

Seems lot of choices for conditioning here. Can you recommend what you want?